src/content/protocols/pendle.md

```diff
@@ -0,0 +1,415 @@
+
+### Core Protocol Components
+
+The `BaseSplitCodeFactoryContract` creates large contracts by splitting their bytecode into two separate fragments that function together as a single logical entity. The resulting fragment contracts remain unverified on Etherscan, creating significant transparency issues as their source code is inaccessible to the public. This contract is controlled by the EOA "Pendle: Deployer 1" and is used to deploy contracts like `PendleMarketFactoryV3`.
+
+The `ProxyAdmin` contract controls all `TransparentUpgradeableProxy` contracts in the protocol through functions including `upgrade`, `upgradeAndCall`, and `changeProxyAdmin`. These functions enable complete replacement of contract logic while preserving state data, with no timelock protection.
+
+The `upgradeToAndCall` function, available across the system's proxy contracts, permits upgrading a contract and executing initialization code atomically in a single transaction. Among all protocol contracts, only the `PENDLE` token implements a timelock mechanism (7 days via `initiateConfigChanges`). All other contracts can be upgraded instantaneously without delay or community review.
+
+### Yield Tokenization Infrastructure
+
+The Yield Tokenization infrastructure presents significant upgradeability risks through multiple interconnected contracts. The `PendleCommonSYFactory`  is controlled by "Pendle: Deployer 1", an EOA. This contract can register arbitrary implementation code for new SY tokens via `setSYCreationCode`, creating a critical vulnerability where malicious code could be deployed for all future yield-bearing assets.
+
+The `PendleYieldContractFactory`, controlled by the `Governance` , can modify economic parameters without timelock via `setInterestFeeRate` and `setTreasury`. These functions allow instant changes to protocol fees (up to 20%) and redirection of revenue streams without warning users, leaving them no opportunity to exit their positions.
+
+The SY tokens themselves `PendleERC4626SYV2` (example SY contract) contain `pause` and `unpause` functions that can be triggered by any address with appropriate permissions in the `PendleGovernanceProxy`. This creates an additional upgradeability risk where user assets could be instantly frozen through governance actions without timelock protection.
+
+### Pendle Swap System
+
+The PendleSwap component employs a custom selector-based router proxy `PendleRouterV4` that directs calls to seven specialized implementations. Unlike standard upgradeable proxies, PendleRouterV4's routing infrastructure is controlled by `ActionStorageV4`, whose owner is permanently set to the zero address, making its configuration immutable and eliminating upgradeability risks for this specific component.
```

Comment:
> The PendleSwap component employs a custom selector-based router proxy `PendleRouterV4`

The `PendleRouterV4` routes all user actions not just swap, but also tokenization